### PR TITLE
TS Strict

### PR DESCRIPTION
**Pull request by zlarsen**
This PR converts the Repo to using TypeScript strict mode. Please note the usage of `FIXME_any`. This was a way to get to strict mode and not change too much. It also makes find and replace easier.

I definitely don't know badmagic as well as I could, so I tested what I can think of, but I am sure there are still some areas that need to be tested to make sure they still work.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
     "noFallthroughCasesInSwitch": true,
     "esModuleInterop": true,
     "lib": ["dom", "es7"],
-    "moduleResolution": "node"
+    "moduleResolution": "node",
+    "strict": true
```
Comment (bjyoungblood) [inline tsconfig.json:15]: This is my favorite thing I've seen all day!

File: src/route/Params.tsx [hunk 2]
```diff
@@ -7,7 +7,13 @@ import Helpers from "../lib/helpers";
 
 import { Route, Param, ParamType } from "../types";
 
-function mapInputs({ inputs, route, paramType, reFetch, parent }) {
+function mapInputs({ inputs, route, paramType, reFetch, parent }: {
```
Comment (bjyoungblood) [inline src/route/Params.tsx:10]: Just a readability suggestion: I usually define a named type for object parameters that have more than 1-2 properties. Something like this:

```
interface MapInputsOptions {
  inputs: FIXME_any[];
  route: Route;
  paramType: ParamType;
  reFetch: () => void;
  parent: FIXME_any;
}

function mapInputs({ inputs, route, paramType, reFetch, parent }: MapInputsOptions) {
  ...
}
```

**Review by bjyoungblood — APPROVED** [review]
You rock for doing this! I can't wait to do a follow-on to start getting rid of `any`s. ✨ 💅 🌈

File: src/route/Params.tsx (hunk 2 above)
Comment (zlarsen) [reply src/route/Params.tsx]: Smart! Updated!

**Review by devin-henslee — APPROVED** [review]
Love it! I went through the types and everything looks good.

**Review by jwdotjs — APPROVED** [review]

**Pull request merged**